src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupService.scala

```diff
@@ -85,10 +86,50 @@ class ManagedGroupService(private val resourceService: ResourceService, private
       _ <- resourceService.deleteResource(Resource(managedGroupType.name, groupId))
     } yield ()
   }
+
+  def listPolicyMemberEmails(resourceId: ResourceId, policyName: ManagedGroupPolicyName): Future[Set[WorkbenchEmail]] = {
+    val resourceAndPolicyName = ResourceAndPolicyName(Resource(ManagedGroupService.managedGroupTypeName, resourceId), policyName.asInstanceOf[AccessPolicyName])
+    accessPolicyDAO.loadPolicy(resourceAndPolicyName) flatMap {
+      case Some(policy) => directoryDAO.loadSubjectEmails(policy.members)
+      case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"Group or policy could not be found: $resourceAndPolicyName"))
+    }
+  }
+
+  def overwritePolicyMemberEmails(resourceId: ResourceId, policyName: ManagedGroupPolicyName, emails: Set[WorkbenchEmail]): Future[AccessPolicy] = {
+    val resourceAndPolicyName = ResourceAndPolicyName(Resource(ManagedGroupService.managedGroupTypeName, resourceId), policyName.asInstanceOf[AccessPolicyName])
+    accessPolicyDAO.loadPolicy(resourceAndPolicyName).flatMap {
+      case Some(policy) => {
+        val updatedPolicy = AccessPolicyMembership(emails, policy.actions, policy.roles)
+        resourceService.overwritePolicy(managedGroupType, resourceAndPolicyName.accessPolicyName, resourceAndPolicyName.resource, updatedPolicy)
+      }
+      case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"Group or policy could not be found: $resourceAndPolicyName"))
+    }
+  }
+
+  def addSubjectToPolicy(resourceId: ResourceId, policyName: ManagedGroupPolicyName, subject: WorkbenchSubject): Future[Unit] = {
+    val resourceAndPolicyName = ResourceAndPolicyName(Resource(ManagedGroupService.managedGroupTypeName, resourceId), policyName.asInstanceOf[AccessPolicyName])
+    resourceService.addSubjectToPolicy(resourceAndPolicyName, subject)
+  }
+
+  def removeSubjectFromPolicy(resourceId: ResourceId, policyName: ManagedGroupPolicyName, subject: WorkbenchSubject): Future[Unit] = {
+    val resourceAndPolicyName = ResourceAndPolicyName(Resource(ManagedGroupService.managedGroupTypeName, resourceId), policyName.asInstanceOf[AccessPolicyName])
+    resourceService.removeSubjectFromPolicy(resourceAndPolicyName, subject)
+  }
 }
 
 object ManagedGroupService {
-  val memberRoleName = ResourceRoleName("member")
   val managedGroupTypeName = ResourceTypeName("managed-group")
   val groupNameRe = "^[A-z0-9_-]+$".r
+  private val memberValue = "member"
+  private val adminValue = "admin"
+
+  // In lieu of an Enumeration, this trait is being used to ensure that we can only have these policies in a Managed Group
```

Comment:
see this for a good way to implement an enum
https://github.com/broadinstitute/workbench-libs/blob/develop/util/src/main/scala/org/broadinstitute/dsde/workbench/util/health/StatusModel.scala#L16

Reply:
I tried the case class approach first when I was doing this, and it didn't work because `AccessPolicyName` is already a case class, which means that I can't extend it.  

If you're still not thrilled with this implementation, maybe we can take a few minutes to look at it together